crates/bevy_core_pipeline/src/core_3d/main_pass_3d_node.rs

```diff
@@ -95,13 +96,16 @@ impl Node for MainPass3dNode {
                 }),
             };
 
-            opaque_phase.render(
-                world,
-                render_context,
-                view_entity,
-                camera.viewport.as_ref(),
-                pass_descriptor,
-            );
+            let render_pass = render_context
+                .command_encoder
+                .begin_render_pass(&pass_descriptor);
+            let mut render_pass = TrackedRenderPass::new(render_pass);
+
+            if let Some(viewport) = camera.viewport.as_ref() {
```

Comment:
I consider losing the "auto camera viewport" stuff is a bit of a loss, as I suspect omitting this will be a common foot-gun / a source of "viewport mismatch" bugs. That being said, I also like that we've made this abstraction less opinionated, as there will be cases where we don't want this.

Not something we should do in this pr, but maybe we could have something like this / prefer its use where possible?

```rust
// This would create and set up a TrackedRenderPass, which would set the viewport, if it is
// configured on the camera. "pass" would need to be a wrapper over TrackedRenderPass.
let pass = camera.begin_render_pass(render_context, view_entity);
pass.render_phase(opaque_phase, world);
```

Reply:
Having another pass wrapper type seems a bit bleh, so its worth trying to find a way to remove the need for that.

Reply:
That is an interesting idea. I like it!
If we want to automatically configure the pass based on the camera, then we will have to supply multiple components or a query, though. 

Why couldn't we use the TrackedRenderPass directly or an extension trait?

I think I will play around with this idea tomorrow.